packages/sn-list-controls-react/src/ContentList/virtualized-table.tsx

```diff
@@ -0,0 +1,305 @@
+/* eslint-disable import/named */
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
+import TableCell from '@material-ui/core/TableCell'
+import { ActionModel, FieldSetting, GenericContent } from '@sensenet/default-content-types'
+import clsx from 'clsx'
+import React, { useCallback, useMemo } from 'react'
+import {
+  AutoSizer,
+  Column,
+  Index,
+  RowMouseEventHandlerParams,
+  Table,
+  TableCellProps,
+  TableCellRenderer,
+} from 'react-virtualized'
+import Checkbox from '@material-ui/core/Checkbox'
+import { TableSortLabel, Tooltip } from '@material-ui/core'
+import { ContentListBaseProps } from './content-list-base-props'
+import { ActionsCell, DateCell, ReferenceCell, RowCheckbox, VirtualDefaultCell, VirtualDisplayNameCell } from '.'
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    flexContainer: {
+      display: 'flex',
+      alignItems: 'center',
+      boxSizing: 'border-box',
+    },
+    tableRow: {
+      cursor: 'pointer',
+    },
+    tableRowHover: {
+      '&:hover': {
+        backgroundColor: theme.palette.type === 'dark' ? 'rgba(255, 255, 255, 0.14)' : 'rgba(0, 0, 0, 0.07)',
+      },
+    },
+    tableCell: {
+      flex: 1,
+    },
+    noClick: {
+      cursor: 'initial',
+    },
+    selected: {
+      backgroundColor: theme.palette.type === 'dark' ? 'rgba(255, 255, 255, 0.08)' : 'rgba(0, 0, 0, 0.04)',
+    },
+    selectedLight: {
+      backgroundColor: 'rgba(255, 255, 255, 0.04)',
+    },
+  }),
+)
+
+interface Row {
+  index: number
+}
+
+export interface VirtualCellProps {
+  tableCellProps: TableCellProps
+  fieldSettings: FieldSetting
+}
+
+interface VirtualizedTableProps<T = GenericContent> extends ContentListBaseProps {
+  /**
+   * Contains custom cell template components
+   */
+  cellRenderer?: (props: VirtualCellProps) => React.ReactNode
+  /**
+   * Array of fields that should be displayed
+   * @default []
```

Comment:
Where do you set this to empty array? Also if it has a default value then it should not be mandatory.